tree/dataframe/test/CMakeLists.txt

```
@@ -44,7 +44,13 @@ if(NOT (MSVC OR (APPLE AND CMAKE_SYSTEM_PROCESSOR MATCHES arm64)) OR win_broken_
   ROOT_ADD_GTEST(dataframe_snapshot dataframe_snapshot.cxx LIBRARIES ROOTDataFrame)
 endif()
 
-ROOT_ADD_GTEST(dataframe_datasetspec dataframe_datasetspec.cxx LIBRARIES ROOTDataFrame)
+if(builtin_nlohmannjson)
+  ROOT_ADD_GTEST(dataframe_datasetspec dataframe_datasetspec.cxx LIBRARIES ROOTDataFrame)
+  target_include_directories(dataframe_datasetspec PRIVATE ${CMAKE_SOURCE_DIR}/builtins)
+else()
+  ROOT_ADD_GTEST(dataframe_datasetspec dataframe_datasetspec.cxx LIBRARIES ROOTDataFrame nlohmann_json::nlohmann_json)
+endif()
```

```suggestion
ROOT_ADD_GTEST(dataframe_datasetspec dataframe_datasetspec.cxx LIBRARIES ROOTDataFrame)
if(builtin_nlohmannjson)
  target_include_directories(dataframe_datasetspec PRIVATE ${CMAKE_SOURCE_DIR}/builtins)
else()
  target_link_libraries(dataframe_datasetspec PRIVATE nlohmann_json::nlohmann_json)
endif()
```
which follows the same pattern used elsewhere (`RIO`, `roofit/multiprocess`, etc.)

---

With the tests cmake complains when I write:
```
target_link_libraries(dataframe_datasetspec PRIVATE nlohmann_json::nlohmann_json)
```
Here `PRIVATE` is problematic. I got error:
```CMake Error at tree/dataframe/test/CMakeLists.txt:51 (target_link_libraries):
  The plain signature for target_link_libraries has already been used with
  the target "dataframe_datasetspec".  All uses of target_link_libraries with
  a target must be either all-keyword or all-plain.

  The uses of the plain signature are here:

   * /home/linev/build/webgui/RootMacros.cmake:1406 (target_link_libraries)
   * /home/linev/build/webgui/RootMacros.cmake:1832 (target_link_libraries)
   * /home/linev/build/webgui/RootMacros.cmake:1834 (target_link_libraries)
```

---

One can use `target_link_libraries(dataframe_datasetspec nlohmann_json::nlohmann_json)`

---

I think that is fine as `dataframe_datasetspec` is an executable and thus `PRIVATE` does not make a difference